### PR TITLE
[PR #934 backport][stable-1] proxmox_kvm: code cleanup

### PR DESCRIPTION
**Pull request by Aversiste**
Backport of #934 to stable-1.

### CODE REVIEW

**Comment by felixfontein** [comment]
@Aversiste could you please do `git cherry-pick -x <commit-sha-from-main-branch>` the next time? We expect backports to be created this way.

**Review by felixfontein — APPROVED** [review]
Confirmed manually that changes are identical (whenever possible).

File: plugins/modules/cloud/misc/proxmox_kvm.py [hunk 1]
```diff
@@ -1152,7 +1152,7 @@ def main():
                 module.exit_json(changed=False)
 
             proxmox_node = proxmox.nodes(vm[0]['node'])
-            if getattr(proxmox_node, VZ_TYPE)(vmid).status.current.get()['status'] == 'running':
+            if vm[0]['status'] == 'running':
                 module.exit_json(changed=False, msg="VM %s is running. Stop it before deletion." % vmid)
             taskid = getattr(proxmox_node, VZ_TYPE).delete(vmid)
```
Comment (felixfontein) [inline plugins/modules/cloud/misc/proxmox_kvm.py:1157]: Ah, almost: this needs to be changed to
```suggestion
            taskid = proxmox_node.qemu.delete(vmid)
```

**Review by felixfontein — CHANGES_REQUESTED** [review]

**Comment by Aversiste** [comment]
> @Aversiste could you please do `git cherry-pick -x <commit-sha-from-main-branch>` the next time? We expect backports to be created this way.

Yes sorry, is it better now ?

File: plugins/modules/cloud/misc/proxmox_kvm.py (hunk 1 above)
Comment (Aversiste) [reply plugins/modules/cloud/misc/proxmox_kvm.py]: Taken care of.

**Review by felixfontein — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@Aversiste thanks a lot for creating the backport!

**Pull request merged**